### PR TITLE
[Ubuntu] Migrate PowerShell and Docker tests to Pester

### PR DESCRIPTION
**Pull request by al-cheb**
# Description
Migrating tests:
- Install-PowerShellModules.ps1
- azpowershell.sh
- powershellcore.sh
- docker-compose.sh
- docker-moby.sh

#### Related issue:
https://github.com/actions/virtual-environments-internal/issues/1578
## Check list
- [x] Related issue / work item is attached
- [x] Tests are written (if applicable)
- [ ] Documentation is updated (if applicable)
- [x] Changes are tested and related VM images are successfully generated

### CODE REVIEW

**Review by maxim-lobanov — APPROVED** [review]

**Review by maxim-lobanov — APPROVED** [review]

**Review by dmitry-shibanov — APPROVED** [review]

File: images/linux/scripts/tests/PowerShellModules.Tests.ps1 [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    }
+
+    if ($withVersionsModules) {
+        It "<moduleName> with <expectedVersion> is installed" -TestCases $withVersionsModules {
+            (Get-Module -Name $moduleName -ListAvailable).Version -contains $expectedVersion | Should -BeTrue
+        }
+    }
+}
+
+Describe "AzureModules" {
```
Comment (AlenaSviridenko) [inline images/linux/scripts/tests/PowerShellModules.Tests.ps1:25]: Maybe split PoweShell and Azure modules to 2 separate files? These modules seem unrelated to each other

Comment (al-cheb) [reply images/linux/scripts/tests/PowerShellModules.Tests.ps1:25]: I don't think we should split on two files, because it's more obvious to store all tests for PowerShell modules in a single place.

**Review by vsafonkin — APPROVED** [review]

**Review by AlenaSviridenko — APPROVED** [review]
